test/integration/integration_test.cc

```diff
@@ -1550,6 +1550,36 @@ TEST_P(IntegrationTest, TestUpgradeHeaderInResponse) {
   EXPECT_EQ("Hello World", response->body());
 }
 
+TEST_P(IntegrationTest, TestUpgradeHeaderInResponseWithTrailer) {
```

Comment:
IIUC the idea is to validate that upgrade is removed if the request is missing upgrade, but the rest of the trailers and unrelated headers are preserved?

Reply:
Yes that's correct! Added a comment.